libmscore/cmd.cpp

```diff
@@ -187,6 +187,8 @@ void Score::endCmd(bool rollback)
             masterScore()->_autosaveDirty = true;
             }
       MuseScoreCore::mscoreCore->endCmd();
+      MuseScoreCore::mscoreCore->updateTimeline();
```

Comment:
Is there some reason to place it here? Score does not seem to be the right object to care about timeline widget.

Reply:
It's there because I wanted a point that would catch all score edits. Since it basically mimics endCmd. If there is a way to specifically call Timeline::updateTimeline() in endCmd only when the score is being updated, please let me know :-)

Reply:
Still it probably doesn't catch changes that originate from undo/redo operations. Unfortunately I don't know the right place to catch score changes and nothing more. But maybe some of these options could work:
- Inherit `Timeline` from [`MuseScoreView`](https://github.com/musescore/MuseScore/blob/83da80734f334f0ff4254fd23a00bc6a5647094e/libmscore/mscoreview.h) and implement `updateAll()`/`dataChanged()`. Semantically Timeline is a view that represents a score so this approach seems to look correctly within the MuseScore architecture.
The view should be registered in a score, `ScoreView` or piano roll editor should do something similar already.
- Track a score state (deduced from undo stack) and update the timeline only if this state changes. This is the approach I used for updating a score diff in a score comparison tool, see [this](https://github.com/musescore/MuseScore/blob/83da80734f334f0ff4254fd23a00bc6a5647094e/libmscore/scorediff.cpp#L820-L821) and [this](https://github.com/musescore/MuseScore/blob/83da80734f334f0ff4254fd23a00bc6a5647094e/libmscore/scorediff.h#L157). Not sure whether this is the best aproach :) But it should also work.

It is possible also to actually implement something in `MuseScoreCore` that satisfies these requirements but maybe the first option above would be more preferable.

Reply:
Also, I can implement undo/redo by simply moving MuseScoreCore::mscoreCore()->updateTimeline() to Score::updateTimeline(). Makes more sense there anyways--possibly not the best solution still.

Reply:
I apologise Joshua because you asked me about this in chat and I didn't have any great answer.

Would it make sense to have MuseScoreView emit a signal emit a signal when there has been a change which triggers an update slot in Timeline?

Reply:
IIRC, you can't use slots and signals outside of Q_OBJECT? So, I'd have to latch in the signal to something. Scoreview? It seems MuseScoreView is used in multiple classes, so I'm not sure.